### PR TITLE
Add properties to set element Id.

### PR DESCRIPTION
**Pull request by rluiten**
Add properties to set Id of elements in Sidebar everywhere you can set class at moment.
Add setting contentId in example.
Add to do README.md documentation of properties.

### CODE REVIEW

**Comment by rluiten** [comment]
I have a had a look at the travis build fail, but its not anything I have added.

File: src/sidebar.js [hunk 1]
```diff
@@ -346,12 +346,17 @@ class Sidebar extends Component {
       }
     }
 
+    if (this.props.rootId) {
```
Comment (markusenglund) [inline src/sidebar.js]: Could you put the rootId in rootProps at line 247 instead so it mirrors rootClassName? I think that would be cleaner.

**Comment by markusenglund** [comment]
This looks good. Could you just fix the thing I pointed out in the review?

**Comment by rluiten** [comment]
Looking at the change now.

**Comment by rluiten** [comment]
Not sure why I didn't do that at first.

File: src/sidebar.js [hunk 2]
```diff
@@ -246,7 +246,8 @@ class Sidebar extends Component {
     const rootProps = {
       className: this.props.rootClassName,
       style: { ...defaultStyles.root, ...this.props.styles.root },
-      role: "navigation"
+      role: "navigation",
+      id: this.props.rootId
```
[the remark this answers is not in the document]
Comment (rluiten) [reply src/sidebar.js]: I have no idea what this result means ?

**Comment by markusenglund** [comment]
Great! I merged it into the 3.0.0 branch which I plan to publish to npm sometime in the next few days.

I don't want merge to master until then, since people might see the README change and assume that the feature already exists on the npm repo.

Thanks!

**Comment by rluiten** [comment]
All good, I have a change setup to submit to DefinitelyTyped to update the types once published.

**Comment by markusenglund** [comment]
@rluiten react-sidebar@3 has been published now! Could you make a PR to definitelyTyped with the types?

**Comment by markusenglund** [comment]
No, that should be all as far as I can tell.

**Comment by rluiten** [comment]
OK have created a PR with DefinitelyTyped for react-sidebar.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/27931.

**Pull request merged**